### DISCUSSION TITLE
Building nullable/non nullable ListArrays

### DISCUSSION THREAD

**Discussion by adriangb** [apache/arrow-rs#3658] [Q&A] [answered]
I’m trying to build ListArrays with various combinations of null ability for the outer and inner items, but it seems that the builder functions hardcode the assumption that it’s nullable (eg here https://github.com/apache/arrow-rs/blob/79bda7d361579cda88fec2eb9b8793ad7f653442/arrow-array/src/builder/generic_list_builder.rs#L137). It seems like I’ll have to drop down to the buffer level, but that’s complicated. Is there an easier way?

**Comment by Jefffrey** [comment, marked as answer]
This is now possible, can use [`GenericListBuilder::with_field()`](https://docs.rs/arrow/latest/arrow/array/struct.GenericListBuilder.html#method.with_field) to override the default field (which sets nullability to true):

https://github.com/apache/arrow-rs/blob/c245a45efef9d9453f121587365e56d53c39d28f/arrow-array/src/builder/generic_list_builder.rs#L298-L301
